com.basistech.m2e.code.quality.checkstyle/src/main/java/com/basistech/m2e/code/quality/checkstyle/MavenPluginConfigurationTranslator.java

```diff
@@ -87,14 +104,41 @@ public boolean isSkip() throws CoreException {
 	}
 
 	public URL getRuleset() throws CheckstylePluginException, CoreException {
-		final URL ruleset = resolveLocation(this.getConfigLocation());
+		String configLocation = this.getConfigLocation();
+		URL ruleset = null;
+		if (CHECKSTYLE_DEFAULT_CONFIG_FILE_NAME.equals(configLocation)) {
+			ruleset = getInlineRules();
+		}
+		if (ruleset == null) {
+			ruleset = resolveLocation(this.getConfigLocation());
+		}
 		if (ruleset == null) {
 			throw new CheckstylePluginException(
 			        "Failed to resolve RuleSet from configLocation,SKIPPING Eclipse checkstyle configuration");
 		}
 		return ruleset;
 	}
 
+	private URL getInlineRules() throws CoreException, CheckstylePluginException {
+		PlexusConfiguration config = getParameterValue("checkstyleRules", XmlPlexusConfiguration.class);
+		if (config == null || ((config = config.getChild(0)) == null)) {
+			return null;
+		}
+		
+		Bundle bundle = FrameworkUtil.getBundle(getClass());
+		IPath stateLocation = Platform.getStateLocation(bundle);
+		File dir = stateLocation.toFile();
```

Comment:
@Serranya where is this? I'm not so much familiar with Eclipse, is it some place meant to host this kind of generated file?

Reply:
It's a folder in the .metadata folder of the current workspace. It is something like this:
`${WORKSPACE}/.metadata/com.basistech.m2e.code.quality.feature/

Plugins can put files they need in there. The M2E plugin for example saves the configured lifecyclemappings in there.

Reply:
It's better to use the Activator instance for this, it has a  `getStateLocation()`

Reply:
Isn't there any way to get the activator instance from the plugin without relying on a singleton?

Reply:
That's the usual way. As the Plugins are marked as Singleton there is only ever a single instance anyways

Reply:
thanks